### PR TITLE
Json path transformation

### PR DESCRIPTION
**Pull request by clinique**

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Thanks, looks good. I'll create the according CQs for the libraries.
As the code mentions Sebastian Janzen as co-author, could you please add an Also-by footer to your commit (and squash them now)? Thanks!

**Comment by kaikreuzer** [comment]
I created CQ 9851 for JsonPath.

Where did you get the other two jars from? There are no official releases at https://github.com/netplex/json-smart-v2/releases, nor are the sources tagged in any way.
I would need a clear reference to the sourcecode of these specific versions that you have included.

**Comment by clinique** [comment]
Links to source repositories are referenced in the about.html.
The jar files comes from Maven repository : 
http://mvnrepository.com/artifact/net.minidev/asm/1.0.2
http://mvnrepository.com/artifact/net.minidev/json-smart/2.1.1
http://mvnrepository.com/artifact/com.jayway.jsonpath/json-path/2.0.0

**Comment by kaikreuzer** [comment]
> Links to source repositories are referenced in the about.html

Right, but they are pointing to the master. How do you know that the current master is exactly the source code for versions 2.1.1 and 1.0.2?

**Comment by clinique** [comment]
Should I instead put in about.html both maven to exact version and reference to master ?

**Comment by kaikreuzer** [comment]
No, the about.html is fine. My problem is that I need to provide a link to the sources to the Eclipse IP team. And who can tell me that there are no new commits in the next days on master, which would make the link invalid? But if you can confirm that the binary on Maven Central has been built with exactly these sources, then I think I can find a solution. But again the question: How do you tell that these were the sources?

**Comment by clinique** [comment]
Unfortunately, I can't swear that jars provided in Maven has been built exactly with these sources. I collect them following the indications provided in the different MANIFEST files of each jar, and also checking that versions of sources and jars were identical.

**Comment by kaikreuzer** [comment]
Ok, I have now also entered CQs 9861 and 9862. Now waiting for approval by the IP team.

**Comment by maggu2810** [comment]
Couldn't we use a specific commit ID and build the jar our self (one time and then bundle them)?

**Comment by kaikreuzer** [comment]
I think we should - at least for the json-smart library.
This has commits from March 2015, see https://github.com/netplex/json-smart-v2/commits/master/json-smart/src/main/java/net/minidev/json
while the version on Maven Central has a timestamp of Dec 2014 (http://mvnrepository.com/artifact/net.minidev/json-smart/2.1.1).
So this definitely does not fit together

File: extensions/transform/org.eclipse.smarthome.transform.jsonpath.test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  <parent>
+    <groupId>org.eclipse.smarthome.transform</groupId>
+    <artifactId>pom</artifactId>
+    <version>0.8.0-SNAPSHOT</version>
+  </parent>
+
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.eclipse.smarthome.transform</groupId>
+  <artifactId>org.eclipse.smarthome.transform.xpath.test</artifactId>
```
Comment (teichsta) [inline extensions/transform/org.eclipse.smarthome.transform.jsonpath.test/pom.xml]: xpath > jsonpath

File: extensions/transform/org.eclipse.smarthome.transform.jsonpath.test/pom.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  <name>Eclipse SmartHome XPath Transformation Service Tests</name>
```
Comment (teichsta) [inline extensions/transform/org.eclipse.smarthome.transform.jsonpath.test/pom.xml]: xpath > jsonpath

**Comment by teichsta** [comment]
please also add the service to org.eclipse.smarthome.feature.runtime.core/feature.xml

**Comment by kaikreuzer** [comment]
@clinique I today received a checkin approval for all 3 CQs (with a small risk that the approval could be withdrawn in future since there are still some outstanding questions from the library authors).

Could you please update this PR by
- solving its merge conflicts
- incorporating @teichsta's comment
- building the JSON Smart lib yourself from the sources (see https://github.com/eclipse/smarthome/pull/350#issuecomment-117652895)?

Thanks!

**Comment by clinique** [comment]
Well, I think I now did it :)

**Comment by kaikreuzer** [comment]
Wonderful, thanks!

**Pull request merged**